### ISSUE TITLE
feature: Customize favicon

### ISSUE THREAD

**Issue by CaKlau** [greenpau/caddy-security#155]
Followup of #153

We are already able to change the logo on the Auth Portal with the `logo` directive. However the favicon is still not customizable.  Can this feature be integrated without great effort?

Add Caddyfile directive:

```
security {
      authentication portal myportal {
            ui {
                logo url /assets/images/image.png
                favicon url /assets/images/image.png   <<< ---  New feature
                static_asset "assets/images/image.png" "image/png" /etc/caddy/assets/image.png
             }
       }
}
```

**Comment by greenpau** [comment]
@CaKlau , i will not be able to prioritize this feature in the next couple of months. The earliest is probably November.

**Comment by ooiuri** [comment]
Hello!

I've been able to use a little workaround to have my custom favicon by using static_asset to replace the used favicon, hope that it helps

```
security {
      authentication portal myportal {
            ui {
                logo url /assets/images/image.png
                static_asset "assets/images/favicon.png" "image/png" /assets/images/new-favicon.png
                static_asset "assets/images/image.png" "image/png" /etc/caddy/assets/image.png
             }
       }
}
```

**Comment by CaKlau** [comment]
@ooiuri  Thanks a lot! 🙂 Works perfectly. 

@greenpau  I think this solution is easy enough and requires no feature request. Shall we close the issue?

**Comment by greenpau** [comment]
@ooiuri, thank you!

> @greenpau I think this solution is easy enough and requires no feature request. Shall we close the issue?

It probably still worth a feature. For example, favicon could be an external image.

**Comment by dczysz** [comment]
It would also be nice to change the file type like we can with `logo url`.  This works but feels wrong:
```
static_asset "assets/images/favicon.png" "image/svg+xml" /path/to/favicon.svg
```

**Comment by greenpau** [comment]
> It would also be nice to change the file type like we can with logo url. This works but feels wrong:

@dczysz , are you saying to do something like this?

`favicon url /assets/images/image.png ”image/png`

**Comment by dczysz** [comment]
Yeah exactly, unless having the mime type on the `static_asset` line is enough, and just have `favicon url` behave similar to `logo url`?

Example scenario: We want to use a PNG logo, but by default you have an SVG logo at /assets/images/logo.svg

**Option 1:**

We can override the `logo url` so the browser requests a PNG and receives a PNG:

```
logo url /assets/images/logo.png
static_asset "assets/images/logo.png" "image/png" /opt/caddy/templates/logo.png
```

**Option 2:**

We can just send a PNG instead when the browser requests the (default) SVG logo:

```
static_asset "assets/images/logo.svg" "image/png" /opt/caddy/templates/logo.png
```

---

These both work just fine, but option 1 seems more "correct" to me so that's what we're using for overriding the logo url.  It would be nice to have the same flexibility with the favicon url as well so if one wants to switch file types (ico, png, svg, etc.) it can be done the more "correct" way.

But like I said, this works just fine doing it the second way, so I'm not sure how big of a deal it really is.  I just figured I'd chime in with our use case while this is still up in the air.
